src/internal/lokiutil/lokiutil.go

```diff
@@ -57,7 +58,16 @@ func QueryRange(ctx context.Context, c *loki.Client, queryStr string, from, thro
 		if err := forEachLine(*resp, func(t time.Time, line string) error {
 			from = t
 			nMsgs++
-			return f(t, line)
+			var entry struct {
```

Comment:
what's the backstory here?

Reply:
So, it looks like in the older version Loki just returned the log line; with the newer version of Loki we get back a structured line which contains the log line as a string. This code parses it out.